ui/app/components/usage/page/index.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@ember/component';
+import { service } from '@ember/service';
+import type ApiService from 'vault/services/api';
+import type { getUsageDataFunction, UsageDashboardData } from '@hashicorp/vault-reporting/types/index';
+import { allEngines } from 'vault/helpers/mountable-secret-engines';
+import { allMethods } from 'vault/helpers/mountable-auth-methods';
+import type FlagsService from 'vault/services/flags';
+export default class ClientsActivityComponent extends Component {
+  @service declare readonly api: ApiService;
+  @service declare readonly flags: FlagsService;
+
+  handleFetchUsageData: getUsageDataFunction = async () => {
+    //TODO: Update client with typed response after the API is updated https://hashicorp.atlassian.net/browse/VAULT-35108
+    const response = await this.api.sys.systemReadUtilizationReport();
+    const data = response.data as UsageDashboardData;
+    // Replace engine names with display names if available
+    allEngines().forEach((engine) => {
+      if (engine.type in data.secret_engines) {
+        data.secret_engines[engine.displayName] = data.secret_engines[engine.type] || 0;
+        delete data.secret_engines[engine.type];
+      }
+    });
+    // Replace auth method names with display names if available
+    allMethods().forEach((method) => {
+      if (method.type in data.auth_methods) {
+        data.auth_methods[method.displayName] = data.auth_methods[method.type] || 0;
+        delete data.auth_methods[method.type];
+      }
+    });
+    return data as UsageDashboardData;
+  };
+
+  get isVaultDedicated(): boolean {
```

Comment:
Because `this.flags` is a service you can call it throughout the app, including hds files. Might save you from having this getter. Also, I might be missing it, but I'm not seeing this getter being consumed. I only see it passed into a component.

Reply:
Yeah I think we can eliminate the getter and pass in the flag service property/getter directly but it is being consumed by the component that we are passing it to.

Reply:
I've removed the getter and gone straight to the flag service.